mono/mini/mini-runtime.c

```diff
@@ -1429,10 +1429,11 @@ mono_resolve_patch_target (MonoMethod *method, MonoDomain *domain, guint8 *code,
 	case MONO_PATCH_INFO_GC_SAFE_POINT_FLAG:
 #if defined(__native_client_codegen__)
 		target = (gpointer)&__nacl_thread_suspension_needed;
-#elif defined (USE_COOP_GC)
-		target = (gpointer)&mono_polling_required;
 #else
-		g_error ("Unsuported patch target");
+		if (mono_threads_is_coop_enabled ())
+			target = (gpointer)&mono_polling_required;
+		else
+			g_error ("Unsuported patch target");
```

Comment:
It would be nice if we could flag an AOT / FullAOT module as using coop/async and fail to load them instead of bombing.